File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    void in(TreeNode root, ArrayList<Integer> res){
        if(root==null) return;
        in(root.left, res);
        res.add(root.val);
        in(root.right, res);
    }
    public List<Integer> inorderTraversal(TreeNode root) {
        ArrayList<Integer> res = new ArrayList<>();
        in(root, res);
        return res;
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        ans =[]
        q = deque([root])
        while q:
            ans.append(q[-1].val)
            print(ans)
            for _ in range(len(q)):
                for a in q: print("q check:",a.val, end=" ")
                print()
                cur = q.popleft()
                print("popped: ", cur.val)
                if cur.left:
                    q.append(cur.left)
                if cur.right:
                    q.append(cur.right)
                for a in q: print("q check 2:",a.val, end=" ")
                print()
                
            
            
        return (ans)
                
        